Jasmine/Common/ViewControllers/PhraseViewController.swift

```diff
@@ -0,0 +1,39 @@
+import UIKit
+import AVFoundation
+
+class PhraseViewController: UIViewController {
+
+    @IBOutlet private weak var hanZiLabel: UILabel!
+    @IBOutlet private weak var pinYinLabel: UILabel!
+    @IBOutlet private weak var englishLabel: UILabel!
+
+    private let synthesizer = AVSpeechSynthesizer()
+
+    private var viewModel: PhraseViewModel!
+
+    @IBAction private func playButtonPressed(_ sender: Any) {
+        synthesizer.speak(viewModel.speech)
+    }
+
+    override func viewDidLoad() {
+        hanZiLabel.text = viewModel.hanZi
+        pinYinLabel.text = viewModel.pinYin
+        englishLabel.text = viewModel.english
+
+        for label in [hanZiLabel, pinYinLabel, englishLabel] {
+            label?.sizeToFit()
+        }
+    }
+
+    /// Injects the required data before opening this view.
+    ///
+    /// - Parameter viewModel: the view model of this class.
+    func segueWith(_ viewModel: PhraseViewModel) {
+        self.viewModel = viewModel
```

Comment:
Decoupling the two means if in the future the Model changes, or View requirement changes, we can use VM to transit the change instead of storing logic in the ViewController.